### ISSUE TITLE
Can't do pg_repack on hypertable

### ISSUE THREAD

**Issue by pgvirtuoso** [timescale/timescaledb#3303]
**Relevant system information:**
 - OS: Ubuntu 18.04.5 LTS (Bionic Beaver)
 - PostgreSQL version: PostgreSQL 12.7 (Ubuntu 12.7-1.pgdg18.04+1) on x86_64-pc-linux-gnu
 - TimescaleDB version (output of `\dx` in `psql`): 2.3.0
 - Installation method: apt install

**Describe the bug**
I installed pg_repack using pgxn command "sudo pgxn install pg_repack". when I execute pg_repack get the following errors:
```
postgres@timescale-dev-01:/home/ubuntu$ /usr/lib/postgresql/12/bin/pg_repack -d prometheus -t <table name>
INFO: repacking table "<table name>"
WARNING: canceling conflicted backends
ERROR: query failed: ERROR:  hypertables do not support  enabling or disabling triggers.
DETAIL: query was: ALTER TABLE <table name> ENABLE ALWAYS TRIGGER repack_trigger
NOTICE:  table "log_6532996" does not exist, skipping
NOTICE:  type "repack.pk_6532996" does not exist, skipping
NOTICE:  trigger "repack_trigger" for relation "<table name>" does not exist, skipping
postgres@timescale-dev-01:/home/ubuntu$
```

I also tried:
```
postgres@timescale-dev-01:/home/ubuntu$ /usr/lib/postgresql/12/bin/pg_repack -d prometheus -I public.vibration_sensor_data
INFO: repacking table "_timescaledb_internal._vibration_sensor_data_1026_chunk"
ERROR: query failed: ERROR:  operation not supported on chunk tables
DETAIL: query was: ALTER TABLE _timescaledb_internal._vibration_sensor_data_1026_chunk ENABLE ALWAYS TRIGGER repack_trigger
NOTICE:  table "log_7733708" does not exist, skipping
NOTICE:  type "repack.pk_7733708" does not exist, skipping
NOTICE:  trigger "repack_trigger" for relation "_timescaledb_internal._vibration_sensor_data_1026_chunk" does not exist, skipping
postgres@timescale-dev-01:/home/ubuntu$
```

**To Reproduce**
1. Disable any compression on the hypertable if any.
2. Make sure you have pg_repack from pgxn. (execute sudo pgxn install pg_repack)
3. execute the commands as described above.

**Expected behavior**
This should repack the hypertable

**Actual behavior**
gives an error and wont repack the hypertable.

**Comment by mkindahl** [comment]
Thanks for an excellent feature request @pgvirtuoso !

We do not support adding triggers on hypertables or chunks since we need them for other purposes. It might, however, be possible to change this and allow triggers but it would require some research to make sure that it does not break anything.

I am actually not entirely sure what the `repack_trigger` is needed for in `pg_repack`. It only returns the new row.

**Comment by jflambert** [comment]
hi @mkindahl is your comment still valid with v2.8.0?

> We do not support adding triggers on hypertables

Because we've been adding before insert triggers on hypertables for a while...

**Comment by andreasscherbaum** [comment]
The `pg_repack` project recently received [this bug report](https://github.com/reorg/pg_repack/issues/448):

```
ERROR: hypertables do not support enabling or disabling triggers.
```

Timescaledb extension: 2.13
pg_repack extension: 1.5.2
PG version: 16.8

Can someone help us understanding why this error still appears, if support for before triggers was added already?

**Comment by mkindahl** [comment]
> The `pg_repack` project recently received [this bug report](https://github.com/reorg/pg_repack/issues/448):
> 
> ```
> ERROR: hypertables do not support enabling or disabling triggers.
> ```
> 
> Timescaledb extension: 2.13 pg_repack extension: 1.5.2 PG version: 16.8
> 
> Can someone help us understanding why this error still appears, if support for before triggers was added already?

Support for adding triggers exist, but looking at this code (from `src/process_utility.c`), the statement `ALTER TABLE ... ENABLE ALWAYS TRIGGER` is not supported.

```c
		case AT_EnableTrig:
		case AT_EnableAlwaysTrig:
		case AT_EnableReplicaTrig:
		case AT_DisableTrig:
		case AT_EnableTrigAll:
		case AT_DisableTrigAll:
		case AT_EnableTrigUser:
		case AT_DisableTrigUser:
			ereport(ERROR,
					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
					 errmsg("hypertables do not support  "
							"enabling or disabling triggers.")));
			/* Break here to silence compiler */
			break;
```

**Comment by jflambert** [comment]
yeah I think the real problem here is that timescaledb still doesn't allow enabling/disabling triggers on hypertables. Adding them (or removing them) is just fine.

**Comment by andreasscherbaum** [comment]
Any plans to add this? Also is there someone from TimescaleDB to whom we (`pg_repack`) can speak to about testing?

**Comment by andreasscherbaum** [comment]
@philkra Who is a good contact to talk about better testing in combination with TimescaleDB?

**Comment by philkra** [comment]
@andreasscherbaum please DM me in the Timescale [community Slack](https://www.timescale.com/community) to discuss a test plan, thank you

**Comment by svenklemm** [comment]
Fixed by https://github.com/timescale/timescaledb/pull/9648
